### DISCUSSION TITLE
Filter partial matches not working with numbers

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#2084] [Q&A] [answered]
For Screen layout() `TD::make('address','Address')->sort()->filter(TD::FILTER_TEXT),`

1. if you search by exact match "123 some place", works
2. if you search by partial match "some place", works
3. if you search by partial match "123", it does not return anything

Why do partial search filters not work when there are numbers included in the string?

**Comment by tabuna** [comment, marked as answer]
This is how [automatic filters](https://orchid.software/en/docs/filters/#automatic-http-filtering-and-sorting) work. You can change this to your own at any time.

```php
// http://example.com/demo?filter[id]=1
$model->where('id', '=', 1)


// http://example.com/demo?filter[id]=1,2,3,4,5
$model->whereIn('id', [1,2,3,4,5]);


// http://example.com/demo?filter[content.ru.name]=dwqdwq
$model->where('content->ru->name', 'like', 'dwqdwq');
```

**Reply by wsdnathan** [reply]
Understood but why if I pass `->filter(TD::FILTER_TEXT)` does it treat certain queries as "like" matching, but then if the TEXT string contains numbers only it treats it differently? I understand why in some cases you would only want exact matching for numbers (partial/LIKE matching doesn't make sense for numbers) but in this case I am passing TD::FILTER_TEXT so shouldn't it assume it's a text string and therefore not do this unexpected logic?

Don't you think it's weird how automatic filters work in this case of searching an address? If I create my own Filter I see that as a solution but when you say "Note. You can use the same filters for different models." I'm not really sure I agree with your solution here being the best. Realistically to get around this issue of multi-faceted automatic filters, you would have to create multiple custom Filters since different models contain different attributes and it's unlikely that these filters are reusable.

**Reply by wsdnathan** [reply]
Noting that there was an update in Orchid v11 which allows type casting the filter to a string, and that is now working in the way I expected, thanks for that update.
